src/modules/ekf2/EKF/ev_height_control.cpp

```diff
@@ -57,10 +57,11 @@ void Ekf::controlEvHeightFusion()
 			if (continuing_conditions_passing) {
 				fuseEvHgt();
 
-				if (isHeightResetRequired()) {
+				const bool reset = (_ev_sample_delayed.reset_counter != _ev_sample_delayed_prev.reset_counter);
+				if (isHeightResetRequired() || reset ) {
 					// All height sources are failing
 					resetHeightToEv();
-					resetVerticalVelocityToZero();
+					resetVerticalVelocityToEv(_ev_sample_delayed);
```

Comment:
Do we need to check for ISFINITE() here? I didn't see any other controls do this, including horizontal velocities/positions. I do believe that NaNs can propagate all the way here from the mavlink_receiver odometry handler, but not sure if there is handling of non-finite state values further down the line.

Reply:
The full answer is that needs to be reviewed all the way through if we want to properly handle x&y vs z odometry with more granularity.

For now how about checking if `isfinite(_ev_sample_delayed.pos(2))` above as one of the continuing_conditions_passing?

Reply:
Alright, I've done that and also checked `isfinite(_ev_sample_delayed.vel(2))` for resetting to EV, falling back to zero if no estimate is available (as the original code did).